### ISSUE TITLE
Enhancement Request: Support for User-Specific Configuration in NetBeans Platform Application

### ISSUE THREAD

**Issue by DawidLokiec** [apache/netbeans#6707]
### Description

**Issue Description:**
I have a Maven- and NetBeans-Platform-based Java Desktop App Project. Currently, the application utilizes a configuration file located in the installation directory (`etc/<appname>.conf`) to set important Java runtime parameters using the `default_options` option. However, I would like to provide users with the ability to set additional Java runtime parameters, such as the scaling option `sun.java2d.uiScale`.

The challenge is that users might not have write access to the installation directory where the `etc/<appname>.conf` file is located. I've discovered that it's possible to place a similar configuration file in the NetBeans user directory (`<NetBeans-User-Directory>/etc/<appname>.conf`), where users have write permissions.

My dilemma is that the default options specified in the default_options of the installation directory's` etc/<appname>.conf` file need to be retained for backward compatibility. Additionally, I'd like to append user-specific options from the <`NetBeans-User-Directory>/etc/<appname>.conf` file to the default_options.

I am uncertain about the best approach to achieve this, and **I am concerned that if I use the user directory configuration file, the NetBeans Platform might no longer consider the installation directory's configuration.**

**Proposed Solution:**
I propose enhancing the NetBeans Platform to support a mechanism for user-specific configuration that can work in conjunction with the installation directory's configuration.

1. **Support User Directory Configuration:**

- Allow the NetBeans Platform to recognize and use configuration files located in the NetBeans user directory (`<NetBeans-User-Directory>/etc/<appname>.conf`).

- Ensure that user-specific configurations are appended to the default options defined in the installation directory's configuration.

2. **Preserve Backward Compatibility:**

- Retain the ability to use the installation directory's configuration (`etc/<appname>.conf`) for specifying default options.
- Ensure that existing installations with configuration in the installation directory continue to work as expected.

3. **Documentation:**

- Update documentation to clearly explain the new user-specific configuration mechanism and how it interacts with the installation directory's configuration.

**Epected Behavior:**
Upon implementing this enhancement, users should be able to set additional options in the <`NetBeans-User-Directory>/etc/<appname>.conf` file, and these options should be appended to the default_options specified in the installation directory's etc/<appname>.conf. The NetBeans Platform should seamlessly use both configurations for the application runtime

### Are you willing to submit a pull request?

No

**Comment by neilcsmith-net** [comment]
Yes, this has been discussed recently in the context of the IDE and all the module system add-opens / add-exports too.

One possible approach might involve using arg files for the JPMS related things.

There's also the complication that these files are treated as a shell script on Linux / macOS, but not on Windows.  Interpreting as a script makes it easier to include the default options.

**Comment by mbien** [comment]
I install VisualVM (which uses the NB platform) directly from the linux repo and add two flags via the start menu shortcut without changing the platform conf file:  `-J-Xmx3g -J-Djdk.gtk.version=2.2`
